sip_assembly/routines_helpers.py

```diff
@@ -98,36 +96,56 @@ def setup_csv_file(filepath):
     return csvwriter
 
 
-def write_rights_row(bag_dir, filenames, rights_statement, csvwriter):
-    """Creates a row in the rights CSV file for each file and rights statement."""
-    for file in filenames:
+def get_rights_rows(bag_dir, file, rights_statement):
+    """Gets rows for each rights statement for a file
+
+    Args:
+        bag_dir (str): path to "objects" directory of bag
+        file (str): filename
+        rights
+
+    """
+    rights_rows = []
+    if len(rights_statement.get('rights_granted')) == 0:
+        rights_row = []
+        rights_row.append(os.path.join(bag_dir, file).lstrip('/'))
+        for basis_value in get_basis_fields(rights_statement):
+            rights_row.append(basis_value)
+        count = 10
+        while count < 15:
+            rights_row.insert(count, '')
+            count += 1
+        rights_rows.append(rights_row)
+    else:
         for rights_granted in rights_statement.get('rights_granted'):
-            csvwriter.writerow(
-                [os.path.join(bag_dir, file).lstrip('/'),
-                 rights_statement.get('rights_basis', '').capitalize(),
-                 rights_statement.get('status', ''),
-                 rights_statement.get(
-                    'determination_date', ''), rights_statement.get(
-                    'jurisdiction', ''),
-                    rights_statement.get(
-                    'start_date', ''), rights_statement.get(
-                    'end_date', ''),
-                    rights_statement.get(
-                    'terms', ''), rights_statement.get(
-                    'citation', ''),
-                    rights_statement.get(
-                    'note', ''), rights_granted.get(
-                    'act', ''),
-                    rights_granted.get(
-                    'restriction', ''), rights_granted.get(
-                    'start_date', ''),
-                    rights_granted.get(
-                    'end_date', ''), rights_granted.get(
-                    'note', ''),
-                    rights_statement.get(
-                    'doc_id_type', ''), rights_statement.get(
-                    'doc_id_value', ''),
-                    rights_statement.get('doc_id_role', '')])
+            rights_row = []
+            rights_row.append(os.path.join(bag_dir, file).lstrip('/'))
+            for basis_value in get_basis_fields(rights_statement):
+                rights_row.append(basis_value)
+            count = 10
+            for grant_value in get_grant_restriction(rights_granted):
+                rights_row.insert(count, grant_value)
+                count += 1
```

Comment:
How about if `get_grant_restriction` returns a list (instead of a tuple) and then you just stick the two together using `+` or `extend`?

Reply:
Also, if `get_grant_restriction` always returns a list with the right number of empty strings, then you don't need this case splitting.